node/bft/src/primary.rs

```diff
@@ -342,12 +342,21 @@ impl<N: Network> Primary<N> {
         // Compute the previous round.
         let previous_round = round.saturating_sub(1);
 
+        // If the current round is 0, return early.
+        ensure!(round > 0, "Round 0 cannot have transaction batches");
+
         // If the current storage round is below the latest proposal round, then return early.
         if round < *lock_guard {
             warn!("Cannot propose a batch for round {round} - the latest proposal cache round is {}", *lock_guard);
             return Ok(());
         }
 
+        // Determine if the current proposal is expired.
+        if round == *lock_guard {
+            warn!("Primary is safely skipping a batch proposal - round {round} already proposed");
```

Comment:
The placement of this check is incorrect. This will prevent the node from re-sending the `proposed_batch` if one already exists.

This should be moved to (at least) after the
` // Ensure the primary has not proposed a batch for this round before.`  check. Because the call to `send_primary_round_to_bft` is needed in some cases to update the storage round.

Reply:
Agree. Should be rebroadcast to improve broadcast reliability and try to update the round as early as possible.